docs/src/components/markdown/Installation.tsx

```diff
@@ -0,0 +1,227 @@
+                <Box>
+                  <Typography
+                    role="presentation"
+                    sx={{
+                      fontWeight: 600,
+                      fontSize: '0.75rem',
+                      mb: '2px',
+                      color: 'text.secondary',
+                    }}
+                  >
+                    {item}
+                  </Typography>
+                  <ToggleButtonGroup
+                    size="small"
+                    key={selectedOptions[index]}
+                    fullWidth
+                    color="primary"
+                    value={selectedOptions[index]}
+                    exclusive
+                    onChange={(event, value) => {
+                      if (value) {
+                        setSelectedOptions((prev) => {
+                          const newValue = [...prev];
+                          newValue[index] = value;
+                          return newValue;
+                        });
+                      }
+                    }}
+                    aria-label={item}
+                  >
+                    {options.map((opt) => {
+                      const label = Array.isArray(opt) ? opt[0] : opt;
+                      const value = Array.isArray(opt) ? opt[1] : opt;
+                      return (
+                        <ToggleButton
+                          key={value}
+                          value={value}
+                          sx={{
+                            width: 'max-content',
+                            minWidth: 64,
+                            py: 0.5,
+                            px: 1,
+                          }}
+                        >
+                          {label}
+                        </ToggleButton>
+                      );
+                    })}
+                  </ToggleButtonGroup>
+                </Box>
+              );
+            }
+            return null;
+          })}
+        </Box>
+      )}
+      <TabsUnstyled selectionFollowsFocus value={manager} onChange={handleChange}>
+        <TabsListUnstyled slots={{ root: StyledTabList }}>
+          {managers.map((item) => (
+            <TabUnstyled
+              slots={{ root: StyledTab }}
+              // @ts-ignore
+              slotProps={{ root: { ownerState } }}
+              key={item}
+              value={item}
+            >
+              {item}
+            </TabUnstyled>
+          ))}
+          <Box sx={{ ml: 'auto' }} />
+        </TabsListUnstyled>
+        {managers.map((item) => (
+          <TabPanelUnstyled
+            slots={{ root: StyledTabPanel }}
+            // @ts-ignore
+            slotProps={{ root: { ownerState } }}
+            key={item}
+            value={item}
+          >
+            <HighlightedCode
+              language="bash"
+              code={`${
+                { yarn: 'yarn add', npm: 'npm install', pnpm: 'pnpm add' }[item]
+              } ${finalInstallation}`}
+            />
```

Comment:
Probably too complicated, but would it be possible to split this into two distinct components:

- `HighlightedCodeWithSelect` similar to `HighlightedCode` but accepts a callback for `code` with the selected state
- `Installation` That reuses the previous one to do what you're currently doing.

The example I've in mind is the [pickers setups](https://mui.com/x/react-date-pickers/getting-started/#code-setup). It is not at all about npm or yarn but has 2 selects to adapt the code. In such a case, we could reuse the `HighlightedCodeWithSelect` to get an up-to-date UI

Reply:
👍 I will take a look at it.